tests/core/http.rs

```diff
@@ -55,6 +55,9 @@ impl Http {
 #[async_trait::async_trait]
 impl HttpIO for Http {
     async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+        // configure wait to simulate the actual real world request.
+        let _ = tokio::time::sleep(tokio::time::Duration::from_millis(10)).await;
+
```

Comment:
let's better do the test changes as a separate pr I'm not sure about them right now

Reply:
done, dropped test related changes, will raise separate issue for this.